File: JS/Decorator Pattern/mocha.js
import BeverageDecorator from "./beverageDecorator.js";

class Mocha extends BeverageDecorator {
  cost() {
    return super.cost() + 0.3;
  }

  description() {
    return super.description() + ", Mocha";
  }
}

export default Mocha;

File: Design-Patterns/Go/Decorator-Pattern/Coffee-Shop/mocha.go
package main

type Mocha struct {
	BeverageDecorator
}

func NewMocha(b Beverage) Beverage {
	return Mocha{BeverageDecorator{Beverage: b}}
}

func (m Mocha) Cost() float64 {
	return m.Beverage.Cost() + 0.30
}

func (m Mocha) Description() string {
	return m.Beverage.Description() + ", Mocha"
}
